test/e2e/performanceprofile/functests/1_performance/irqbalance.go

```diff
@@ -0,0 +1,247 @@
+				_, err = util.WaitForCmdInPod(5*time.Second, 5*time.Minute, pod, "test", "-e", "/run/tuned/tuned.pid")
+				return err
+			}).WithTimeout(5 * time.Minute).WithPolling(10 * time.Second).ShouldNot(HaveOccurred())
+
+			By(fmt.Sprintf("re-verifying worker node %q after TuneD restart", targetNode.Name))
+			postRestartBannedCPUs, err := getIrqBalanceBannedCPUs(targetNode)
+			Expect(err).ToNot(HaveOccurred(), "failed to extract the banned CPUs from node %q", targetNode.Name)
+			testlog.Infof("banned CPUs on %q after the tuned restart: {%s}", targetNode.Name, postRestartBannedCPUs.String())
+
+			Expect(postRestartBannedCPUs.ToSlice()).To(Equal(postCreateBannedCPUs.ToSlice()), "banned CPUs changed post tuned restart on node %q", postRestartBannedCPUs.ToSlice(), targetNode.Name)
+		})
+	})
+})
+
+// nodes.BannedCPUs fails (!!!) if the current banned list is empty because, deep down, ExecCommandOnNode expects non-empty stdout.
+// In turn, we do this to at least have a chance to detect failed commands vs failed to execute commands (we had this issue in
+// not-so-distant past, legit command output lost somewhere in the communication). Fixing ExecCommandOnNode isn't trivial and
+// require close attention. For the time being we reimplement a form of nodes.BannedCPUs which can handle empty ban list.
+func getIrqBalanceBannedCPUs(node *corev1.Node) (cpuset.CPUSet, error) {
+	cmd := []string{"cat", "/rootfs/etc/sysconfig/irqbalance"}
+	conf, err := nodes.ExecCommandOnNode(cmd, node)
+	if err != nil {
+		return cpuset.NewCPUSet(), err
+	}
+
+	keyValue := findIrqBalanceBannedCPUsVarFromConf(conf)
+	if len(keyValue) == 0 {
+		// can happen: everything commented out (default if no tuning ever)
+		testlog.Warningf("cannot find the CPU ban list in the configuration (\n%s)\n", conf)
+		return cpuset.NewCPUSet(), nil
+	}
+
+	testlog.Infof("banned CPUs setting: %q", keyValue)
+
+	items := strings.FieldsFunc(keyValue, func(c rune) bool {
+		return c == '='
+	})
+	if len(items) != 2 {
+		return cpuset.NewCPUSet(), fmt.Errorf("malformed CPU ban list in the configuration")
+	}
+
+	bannedCPUs := unquote(strings.TrimSpace(items[1]))
+	testlog.Infof("banned CPUs: %q", bannedCPUs)
+
+	banned, err := components.CPUMaskToCPUSet(bannedCPUs)
+	if err != nil {
+		return cpuset.NewCPUSet(), fmt.Errorf("failed to parse the banned CPUs: %v", err)
+	}
+
+	return banned, nil
+}
+
+func getIrqDefaultSMPAffinity(node *corev1.Node) (string, error) {
+	cmd := []string{"cat", "/rootfs/proc/irq/default_smp_affinity"}
+	return nodes.ExecCommandOnNode(cmd, node)
+}
+
+func findIrqBalanceBannedCPUsVarFromConf(conf string) string {
+	scanner := bufio.NewScanner(strings.NewReader(conf))
+	for scanner.Scan() {
+		line := strings.TrimSpace(scanner.Text())
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+		if !strings.HasPrefix(line, "IRQBALANCE_BANNED_CPUS") {
+			continue
+		}
+		return line
+	}
+	return ""
+}
+
+func pickNodeIdx(nodes []corev1.Node) int {
```

Comment:
consider moving to utils , not mandatory.

Reply:
Intentionally avoided because https://dave.cheney.net/2019/01/08/avoid-package-names-like-base-util-or-common
and I don't have enough material yet to start a new package